### PR TITLE
Storybook: Created Portfolio Tab View

### PR DESCRIPTION
**Pull request by Douglashdaniel**
Resolves <https://github.com/brave/brave-browser/issues/15123>

Created the Portfolio Tab View and added it too storybook under Concepts.
Also needed to add 2 new dependancies from "Recharts" to create the line chart component.

Package: https://www.npmjs.com/package/recharts
Typescript Package: https://www.npmjs.com/package/@types/recharts

Will need to run `npm install`

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/handbook/blob/master/development/security.md#when-is-a-security-review-needed), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/handbook/blob/master/development/security.md#when-is-a-security-review-needed), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

1. `npm run storybook`
2. Confirm that storybook shows the new component.
3. Check functionality of the new component.

https://user-images.githubusercontent.com/40611140/113927389-a695c600-97aa-11eb-8596-b6be23706e5c.mov

### CODE REVIEW

File: package.json
Comment (petemill) [inline package.json:293]: I'm ok with this package in general since it's not adding any deps to an existing WebUI that would have performance concerns. But, before you merge, please could you request a basic security review on the security repo? It's a team policy, just in case there're any red flags. If you don't have access to this repo just let me know and I'll get it started.

**Review by petemill — APPROVED** [review]
Nice work. I have no concerns just queries and nits, aside from getting the security review done for the new dependency. I like the move towards function components and React Hooks that some of this seems to have - small components don't really need to be PureComponents, and function components could be wrapped with React.memo if we wanted them to be super-efficient.

File: components/brave_wallet_ui/assets/asset-icons/zrx-asset-icon.svg
Comment (Douglashdaniel) [reply components/brave_wallet_ui/assets/asset-icons/zrx-asset-icon.svg:1]: Gotcha, did not run through any optimizer. I Just exported from Figma and reviewed, seemed like Figma did a pretty good job! But may be redundant to have both in there. Ultimately it would be awesome to have a lot of these icons in the brave-ui library in the future :)

File: components/brave_wallet_ui/constants/types.ts
Comment (Douglashdaniel) [reply components/brave_wallet_ui/constants/types.ts:9]: lol true! I honestly find it clearer/helpful to name them that way when using throughout a project.

File: package.json
Comment (Douglashdaniel) [reply package.json:293]: https://github.com/brave/security/issues/408

File: components/brave_wallet_ui/components/desktop/views/portfolio/index.tsx
Comment (Douglashdaniel) [reply components/brave_wallet_ui/components/desktop/views/portfolio/index.tsx:20]: Gotcha, will keep that in mind in the future!

File: components/brave_wallet_ui/components/desktop/line-chart/index.tsx
Comment (Douglashdaniel) [reply components/brave_wallet_ui/components/desktop/line-chart/index.tsx:25]: Had to construct it myself :) Basically is just a little pulsating dot for the chart. I'll have another PR soon that will involve this component, i'll make sure to ad some comments in that one.

File: components/brave_wallet_ui/components/desktop/chart-control-bar/style.ts
Comment (Douglashdaniel) [reply components/brave_wallet_ui/components/desktop/chart-control-bar/style.ts:18]: Partial works good for when you only need to use 1 prop and not all from the interface! I think i may have had multiple at one point in time and reduced down to 1.

File: components/brave_wallet_ui/assets/asset-icons/index.ts
Comment (Douglashdaniel) [reply components/brave_wallet_ui/assets/asset-icons/index.ts:1]: Good catch, will do a refactor!

**Pull request merged**
